online_editor/views.py

```diff
@@ -16,19 +16,46 @@ def index(request):
     context = {}
     return render(request, 'index.html', context)
 
-def execute(request):
-    code = request.body.decode() + "\n" #decoding byte to code
-    
+def run(request):
+    code = request.body.decode() + "\n"
+
     input_name = settings.INPUT_FILE_PATH
     output_name = settings.OUTPUT_FILE_PATH
     shell_script = settings.RUN_IN_DOCKER_SH_PATH
-    
+    if request.POST['compile_status'] != True:
+        response = {
+            "message": "The compilation is not complete yet"
+        }
+        return JsonResponse(response, safe=False)
+
+    run_data_frontend = {
+        'source': code,
+        'memory_limit': 243232,
+        'time_limit': 5,
+        'compile_status': True,
+        'lang': "python",
+        'input': "",
+        'id': "213121",
```

Comment:
This "id" entity can be given by front-end, you can get it from the data front-end sent to back-end